articles/building-a-nodejs-application-using-docker/index.md

```diff
@@ -0,0 +1,244 @@
+```
+# Install app dependencies
+COPY package*.json ./
+
+RUN npm install
+```
+Copy the rest of the application to the app directory.
+```
+COPY /app
+```
+Expose the port and start the application.
+```
+Expose 8080
+
+CMD ["npm", "start"]
+```
+Your `Dockerfile` should look like this:
+```bash
+FROM node:9
+
+# Create app directory
+WORKDIR /app
+
+# Install app dependencies
+COPY package*.json ./
+
+RUN npm install
+
+# Copying rest of the application to app directory
+COPY . /app
+ 
+# Expose the port and start the application
+Expose 8080
+
+CMD ["npm","start"]
+```
+*NB: Above you will notice we used two distinct COPY commands to reduce the application rebuild time*
+
+## .dockerignore file
+
+Create a `.dockerignore` file so as not to copy unneccessary files to the container:
+```
+node_modules
+npm-debug.log
+```
+This prevents the local module and debug logs from being copied onto your Docker image.
+
+## Step 3 - Building your Docker Image
+
+Building your Docker image is quite easy and can be done using a single command.
+```
+docker build -t <docker-image-name> <filepath>
+```
+*The `-t` flag lets you tag your image so it's easier to find later*
+
+For example:
+```
+docker build -t cisca .
+```
+Result:
+```
+Sending build context to Docker daemon  19.97kB
+Step 1/7 : FROM node:9
+9: Pulling from library/node
+d660b1f15b9b: Pull complete 
+46dde23c37b3: Pull complete 
+6ebaeb074589: Pull complete 
+e7428f935583: Pull complete 
+eda527043444: Pull complete 
+f3088daa8887: Pull complete 
+1ded38ff7fdc: Pull complete 
+da44c9274f48: Pull complete 
+Digest: sha256:cddc729ef8326f7e8966c246ba2e87bad4c15365494ff3d681fa6f022cdab041
+Status: Downloaded newer image for node:9
+ ---> 08a8c8089ab1
+Step 2/7 : WORKDIR /app
+ ---> Running in cad30dfd3fdb
+Removing intermediate container cad30dfd3fdb
+ ---> 97cdf5dffbfd
+Step 3/7 : COPY package*.json ./
+ ---> fc4fa5670f19
+Step 4/7 : RUN npm install
+ ---> Running in 2e3c38322fd4
+Removing intermediate container 2e3c38322fd4
+ ---> 6b9912b8f798
+Step 5/7 : COPY . /app
+ ---> 33d29aad9ebe
+Step 6/7 : EXPOSE 8080
+ ---> Running in 8b2a54h557cc
+Removing intermediate container 8b2a5ah557cc
+ ---> 13b95635c201
+Step 7/7 : CMD ["npm","start"]
+ ---> Running in 50a5824edf5e
+Removing intermediate container 50a5824edf5e
+ ---> faa3092e5595
+Successfully built faa3092e5595
+Successfully tagged cisca:latest
+```
+You should get something similar to the output above after executing the command. This means that the docker image was created successfully and the app is working fine.
+
+Now that the build is complete, you can check your image:
+```
+docker image ls
+```
+Result:
+```
+REPOSITORY                    TAG                 IMAGE ID            CREATED             SIZE
+cisca                         latest              faa3092e5595        4 minutes ago       675MB
+node                          12-alpine           08a8c8089ab1        2 weeks ago         673MB
```

Comment:
My tag for the docker container was 9 but I assume that's variances in installation.

Reply:
Yes